In [1]:
import os
import pandas as pd
from PIL import Image
import requests

In [2]:
data = pd.read_csv("USvideos.csv") #Download from https://www.kaggle.com/datasnaek/youtube-new/version/113#USvideos.csv

In [3]:
data.head()

video_id trending_date  \
0  2kyS6SvSYSE      17.14.11   
1  1ZAPwfrtAFY      17.14.11   
2  5qpjK5DgCt4      17.14.11   
3  puqaWrEC7tY      17.14.11   
4  d380meD0W0M      17.14.11   

                                               title          channel_title  \
0                 WE WANT TO TALK ABOUT OUR MARRIAGE           CaseyNeistat   
1  The Trump Presidency: Last Week Tonight with J...        LastWeekTonight   
2  Racist Superman | Rudy Mancuso, King Bach & Le...           Rudy Mancuso   
3                   Nickelback Lyrics: Real or Fake?  Good Mythical Morning   
4                           I Dare You: GOING BALD!?               nigahiga   

   category_id              publish_time  \
0           22  2017-11-13T17:13:01.000Z   
1           24  2017-11-13T07:30:00.000Z   
2           23  2017-11-12T19:05:24.000Z   
3           24  2017-11-13T11:00:04.000Z   
4           24  2017-11-12T18:01:41.000Z   

                                                tags    views   likes  \
0                                    SHANtell martin   748374   57527   
1  last week tonight trump presidency|"last week ...  2418783   97185   
2  racist superman|"rudy"|"mancuso"|"king"|"bach"...  3191434  146033   
3  rhett and link|"gmm"|"good mythical morning"|"...   343168   10172   
4  ryan|"higa"|"higatv"|"nigahiga"|"i dare you"|"...  2095731  132235   

   dislikes  comment_count                                  thumbnail_link  \
0      2966          15954  https://i.ytimg.com/vi/2kyS6SvSYSE/default.jpg   
1      6146          12703  https://i.ytimg.com/vi/1ZAPwfrtAFY/default.jpg   
2      5339           8181  https://i.ytimg.com/vi/5qpjK5DgCt4/default.jpg   
3       666           2146  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
4      1989          17518  https://i.ytimg.com/vi/d380meD0W0M/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  SHANTELL'S CHANNEL - https://www.youtube.com/s...  
1  One year after the presidential election, John...  
2  WATCH MY PREVIOUS VIDEO ▶ \n\nSUBSCRIBE ► http...  
3  Today we find out if Link is a Nickelback amat...  
4  I know it's been a while since we did this sho...

In [4]:
data.describe()

category_id         views         likes      dislikes  comment_count
count  34949.000000  3.494900e+04  3.494900e+04  3.494900e+04   3.494900e+04
mean      20.104180  1.880284e+06  6.090286e+04  3.077440e+03   7.017535e+03
std        7.514496  5.756655e+06  1.793106e+05  2.968038e+04   3.061650e+04
min        1.000000  5.490000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       17.000000  1.994290e+05  4.434000e+03  1.680000e+02   5.210000e+02
50%       24.000000  5.557580e+05  1.499700e+04  5.250000e+02   1.577000e+03
75%       25.000000  1.487513e+06  4.478500e+04  1.591000e+03   4.652000e+03
max       43.000000  1.493761e+08  3.388289e+06  1.674420e+06   1.361580e+06

# Download photos

In [5]:
def getPhoto(dat):
    vId = dat[0]
    link = dat[1]
    if not os.path.exists("data//" + vId + ".jpg"):
        r = requests.get(link)
        with open("data//" + vId + ".jpg", "wb+") as f:
            f.write(r.content)
        return 1
    else:
        return 0

In [6]:
dat = [dat for _, dat in data[["video_id","thumbnail_link"]].iterrows()]

In [7]:
tot = len(dat)
c = 0
d_count = 0
s_count = 0

for _, d in enumerate(dat):
    print("Percent Complete: {:.1f}%".format((c/tot)*100), end="\r")
    
    if getPhoto(d):
        d_count += 1
    else:
        s_count += 1
    c+=1

print("\nFinished")
print(str(d_count), "images downloaded")
print(str(s_count), "images skipped. This could be due to an error in the link or the image\
 may have already been downloaded")

Percent Complete: 100.0%
Finished
164 images downloaded
34785 images skipped. This could be due to an error in the link or the image may have already been downloaded


# Some photos are empty photos (see example below) so we need to get rid of those

![title](assets/imgs/empty.jpg)

In [8]:
def getEmptyImgs():
    empty_imgs = []
    for top, d, files in os.walk("data"):
        tot = len(files)
        c = 0
        for file in files:
            print("Percent Complete: {:.1f}%".format((c/tot)*100), end="\r")
            img_samp = Image.open("data/_suPAag6_i8.jpg")
            img_name = "data" +  "//" + file
            tmp = Image.open(img_name) 

            if tmp == img_samp:
                empty_imgs.append(file)

            c+=1
        break
    print("\nFinished")
    print("{} empty images".format(len(empty_imgs)))
    return empty_imgs

In [9]:
empty = [file.replace(".jpg", "") for file in getEmptyImgs()]

Percent Complete: 100.0%
Finished
164 empty images


# Now we want to clean the Dataframe so that we aren't looking at the videos with empty images

In [10]:
data_cleaned = data.loc[~data["video_id"].isin(empty)]

### Delete those images so we don't read them in later

In [11]:
for file in empty:
    f = "data/" + file + ".jpg"
    try:
        os.remove(f)
    except:
        print("Could not find file {}!".format(f))
    else:
        print(file, "DELETED")

-hg9t3PdhFE DELETED
-QR-TB_k20M DELETED
-xt_j2aIAkg DELETED
0-yiRIWkhLo DELETED
01AEuxSlIMg DELETED
04aPsalAZQI DELETED
07JQ4WZJIbg DELETED
0cZLo3AAan4 DELETED
0DVNtFA-_wA DELETED
0EQuVfP-vXs DELETED
1Apnffv9Yko DELETED
1Nu_oGucpDI DELETED
1qKBNKtzTWg DELETED
1VJixjI_P3s DELETED
2B85lYenaxE DELETED
2omJBSQcm5g DELETED
3ekEfYNQJ1s DELETED
3ILxrRlf3KY DELETED
4BiWtH0HywY DELETED
4eKYR_iL5eU DELETED
4HSul1NBFes DELETED
4sXyLTk0IAQ DELETED
51cbHTKdILA DELETED
5JEI0sjJs8c DELETED
5n32F7mS6Kk DELETED
6JvRVILczps DELETED
6ULZDwevR0M DELETED
6VoJHrdoIJw DELETED
7Bq_H1ci9ec DELETED
7uvxjNut1gk DELETED
7YibRBd46oo DELETED
8Ojf1RVqRn4 DELETED
8WbxbqwdWog DELETED
90uhxbfUfC4 DELETED
9TUBf6l7FBg DELETED
9vlKOFd73XM DELETED
9XSULcLI6Hg DELETED
aAm6AHr14zY DELETED
abeqEEmpbdM DELETED
AEAHRq62p3Q DELETED
ai6NByMmCVU DELETED
aKPz583WRpQ DELETED
A_mlvG_nRsg DELETED
bhoFXkVy8JA DELETED
bJje-B9YLIU DELETED
bLKd4XXUWUo DELETED
BLOIU9DpLq4 DELETED
BLXLPisBe8s DELETED
Bmn2STJUWJs DELETED
bT9HPXKP5dI DELETED


# Now for the fun model stuff

### We'll be using a GAN that looks something like:

![title](assets/imgs/model.jpg)

Our model will take in as input:

- The title of the video

- The channel title

- The category id

- Any tags associated to the video


 - Optionally, it may be interesting to look at including publish time

The model will then output a thumbnail image for us to use for our video

### Prepare the Data!

For us to use the above in our model, we need to think of how to vectorize them:

#### Title

    Since the title is a group of words. Lets first take a deeper look at what we're dealing with.

In [12]:
titles = data_cleaned.title

In [13]:
titles

0                       WE WANT TO TALK ABOUT OUR MARRIAGE
1        The Trump Presidency: Last Week Tonight with J...
2        Racist Superman | Rudy Mancuso, King Bach & Le...
3                         Nickelback Lyrics: Real or Fake?
4                                 I Dare You: GOING BALD!?
5                                    2 Weeks with iPhone X
6                Roy Moore & Jeff Sessions Cold Open - SNL
7                      5 Ice Cream Gadgets put to the Test
8        The Greatest Showman | Official Trailer 2 [HD]...
9        Why the rise of the robots won’t mean the end ...
10       Dion Lewis' 103-Yd Kick Return TD vs. Denver! ...
11       (SPOILERS) 'Shiva Saves the Day' Talked About ...
12              Marshmello - Blocks (Official Music Video)
13                  Which Countries Are About To Collapse?
14                                SHOPPING FOR NEW FISH!!!
15                                        The New SpotMini
16        One Change That Would Make Pacific Rim a Class

In [14]:
titles.describe()

count                                                 34223
unique                                                 5859
top       The Chainsmokers, Drew Love - Somebody (Offici...
freq                                                     20
Name: title, dtype: object

##### Hmmm... that's interesting. There are only 5,859 unique titles? Let's dig deeper, shall we?

In [15]:
data_cleaned.loc[titles == "Nickelback Lyrics: Real or Fake?"]  #Title was included in trending 
                                                                #as of 7/11/2018 at least

video_id trending_date                             title  \
3     puqaWrEC7tY      17.14.11  Nickelback Lyrics: Real or Fake?   
231   puqaWrEC7tY      17.15.11  Nickelback Lyrics: Real or Fake?   
471   puqaWrEC7tY      17.16.11  Nickelback Lyrics: Real or Fake?   
709   puqaWrEC7tY      17.17.11  Nickelback Lyrics: Real or Fake?   
946   puqaWrEC7tY      17.18.11  Nickelback Lyrics: Real or Fake?   
1174  puqaWrEC7tY      17.19.11  Nickelback Lyrics: Real or Fake?   
1399  puqaWrEC7tY      17.20.11  Nickelback Lyrics: Real or Fake?   

              channel_title  category_id              publish_time  \
3     Good Mythical Morning           24  2017-11-13T11:00:04.000Z   
231   Good Mythical Morning           24  2017-11-13T11:00:04.000Z   
471   Good Mythical Morning           24  2017-11-13T11:00:04.000Z   
709   Good Mythical Morning           24  2017-11-13T11:00:04.000Z   
946   Good Mythical Morning           24  2017-11-13T11:00:04.000Z   
1174  Good Mythical Morning           24  2017-11-13T11:00:04.000Z   
1399  Good Mythical Morning           24  2017-11-13T11:00:04.000Z   

                                                   tags   views  likes  \
3     rhett and link|"gmm"|"good mythical morning"|"...  343168  10172   
231   rhett and link|"gmm"|"good mythical morning"|"...  772235  15301   
471   rhett and link|"gmm"|"good mythical morning"|"...  827546  15887   
709   rhett and link|"gmm"|"good mythical morning"|"...  863397  16213   
946   rhett and link|"gmm"|"good mythical morning"|"...  885011  16396   
1174  rhett and link|"gmm"|"good mythical morning"|"...  897162  16551   
1399  rhett and link|"gmm"|"good mythical morning"|"...  913268  16729   

      dislikes  comment_count                                  thumbnail_link  \
3          666           2146  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
231       1229           3328  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
471       1299           3460  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
709       1337           2913  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
946       1363           2956  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
1174      1374           2966  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   
1399      1386           2988  https://i.ytimg.com/vi/puqaWrEC7tY/default.jpg   

      comments_disabled  ratings_disabled  video_error_or_removed  \
3                 False             False                   False   
231               False             False                   False   
471               False             False                   False   
709               False             False                   False   
946               False             False                   False   
1174              False             False                   False   
1399              False             False                   False   

                                            description  
3     Today we find out if Link is a Nickelback amat...  
231   Today we find out if Link is a Nickelback amat...  
471   Today we find out if Link is a Nickelback amat...  
709   Today we find out if Link is a Nickelback amat...  
946   Today we find out if Link is a Nickelback amat...  
1174  Today we find out if Link is a Nickelback amat...  
1399  Today we find out if Link is a Nickelback amat...

##### Well that explains that. Since they are trending videos, they can trend for multiple days in a row (or multiple times)

Let's remove any duplicates using the video_id

In [16]:
data_cleaned = data_cleaned.drop_duplicates(subset="video_id")

In [17]:
data_cleaned.loc[titles == "Maroon 5 - Wait"]

video_id trending_date            title channel_title  category_id  \
12604  TnCa71iUVxA      18.18.01  Maroon 5 - Wait   Maroon5VEVO           10   
16983  4uTNVumfm84      18.09.02  Maroon 5 - Wait   Maroon5VEVO           10   

                   publish_time  \
12604  2018-01-17T17:00:03.000Z   
16983  2018-02-08T14:00:00.000Z   

                                                    tags    views   likes  \
12604  Maroon|"Wait"|"222"|"Records/Interscope"|"Reco...   692807   70211   
16983  Maroon|"Wait"|"222"|"Records/Interscope"|"Reco...  2455067  197019   

       dislikes  comment_count  \
12604      2689           5205   
16983      3139          11356   

                                       thumbnail_link  comments_disabled  \
12604  https://i.ytimg.com/vi/TnCa71iUVxA/default.jpg              False   
16983  https://i.ytimg.com/vi/4uTNVumfm84/default.jpg              False   

       ratings_disabled  video_error_or_removed  \
12604             False                   False   
16983             False                   False   

                                             description  
12604  RED PILL BLUES” is out now.\nhttp://smarturl.i...  
16983  RED PILL BLUES” is out now.\nhttp://smarturl.i...

##### That's better. Now let's see that summary again

In [18]:
data_cleaned.describe()

category_id         views         likes       dislikes  comment_count
count  5770.000000  5.770000e+03  5.770000e+03    5770.000000    5770.000000
mean     20.501560  7.043178e+05  3.179135e+04    1324.864298    4156.372270
std       7.221629  1.853097e+06  1.027891e+05   12278.684441   19935.324153
min       1.000000  5.490000e+02  0.000000e+00       0.000000       0.000000
25%      17.000000  7.712800e+04  1.798500e+03      68.000000     250.000000
50%      24.000000  2.465345e+05  7.472500e+03     223.000000     863.000000
75%      25.000000  6.904878e+05  2.326325e+04     700.000000    2601.000000
max      43.000000  4.843165e+07  2.055137e+06  629120.000000  733373.000000

So deduplicating signicantly decreased our available data to train on. No big deal... hopefully.

*continues exploring*

In [19]:
titles = pd.DataFrame(data_cleaned.title)
titles.describe()
# This isn't that bad given that music videos like Maroon 5's can release 2 sometimes. One initial and then a music video.
# They have different video id's and thumbnails so I'll take it

title
count              5770
unique             5764
top     Maroon 5 - Wait
freq                  2

In [20]:
titles["title length"] = titles.title.apply(len)

In [21]:
def getSentenceLen(sent):
    words = sent.split(" ")
    c = 0
    c += sum([1 for word in words])
    return c

In [22]:
titles["word count"] = titles.title.apply(getSentenceLen)

In [23]:
titles

title  title length  \
0                     WE WANT TO TALK ABOUT OUR MARRIAGE            34   
1      The Trump Presidency: Last Week Tonight with J...            62   
2      Racist Superman | Rudy Mancuso, King Bach & Le...            53   
3                       Nickelback Lyrics: Real or Fake?            32   
4                               I Dare You: GOING BALD!?            24   
5                                  2 Weeks with iPhone X            21   
6              Roy Moore & Jeff Sessions Cold Open - SNL            41   
7                    5 Ice Cream Gadgets put to the Test            35   
8      The Greatest Showman | Official Trailer 2 [HD]...            65   
9      Why the rise of the robots won’t mean the end ...            53   
10     Dion Lewis' 103-Yd Kick Return TD vs. Denver! ...            86   
11     (SPOILERS) 'Shiva Saves the Day' Talked About ...            78   
12            Marshmello - Blocks (Official Music Video)            42   
13                Which Countries Are About To Collapse?            38   
14                              SHOPPING FOR NEW FISH!!!            24   
15                                      The New SpotMini            16   
16      One Change That Would Make Pacific Rim a Classic            48   
17     How does your body know you're full? - Hilary ...            52   
18                            HomeMade Electric Airplane            26   
19              Founding An Inbreeding-Free Space Colony            40   
20                      How Can You Control Your Dreams?            32   
21     The Making of Hela's Headdress from Thor: Ragn...            50   
22     Is It Dangerous To Talk To A Camera While Driv...            50   
23     What $4,800 Will Get You In NYC | Sweet Digs H...            67   
24                          Using Other People's Showers            28   
25                SPAGHETTI BURRITO VS SPAGHETTI BURRITO            38   
26                  78557 and Proth Primes - Numberphile            36   
27       A Smart... MUG?! - Take apart a Heated Thermos!            47   
28     LeBron James admits he was ripping Phil Jackso...           100   
29                               Nick Andopolis: Drummer            23   
...                                                  ...           ...   
34556                  Make a Glowing Announcement Board            33   
34558    Livestream Day 1: Amphitheater (Google I/O '18)            47   
34750  Why I'm So Scared (being myself and crying too...            52   
34751  YoungBoy Never Broke Again Goes Sneaker Shoppi...            61   
34753  周杰倫 Jay Chou【不愛我就拉倒 If You Don't Love Me, It's...            63   
34754  Terry Crews Answers the Web's Most Searched Qu...            61   
34755        Why Stradivarius violins are worth millions            43   
34757   Sarah Paulson Gets Scared During '5 Second Rule'            48   
34758  Gabby Barrett Sings I Have Nothing by Whitney ...            89   
34759               The ULTIMATE $30,000 Gaming PC Setup            36   
34760  Rooster Teeth Animated Adventures - Millie So ...            53   
34761  Why You Should Wake Up at 4:30 AM Every Day, A...            69   
34765  AMAZING All You Can Eat  BBQ KEBAB Buffet in I...            51   
34766      Exciting iPhone 11 Leaks & Round Apple Watch!            45   
34768  New lava fissures fuel fears of eruption in Ha...            50   
34769                            Mother Knows Best - SNL            23   
34770                       $1000 Survival Kit in a Case            28   
34771  NEW AMSTERDAM | Official Trailer | NBC Fall Sh...            54   
34772  MANIFEST | Official Trailer | NBC Fall Shows 2018            49   
34774       10 Most DEVASTATING Cyber Attacks in History            44   
34776  Kelly Clarkson - Meaning of Life [Official Video]            49   
34777  We built the Infinity Gauntlet with 25,000 mag...            51   
34781  ‘A Change Is Gonna Come’ for Lifford after he ..

In [24]:
titles.describe()

title length   word count
count   5770.000000  5770.000000
mean      49.531716     8.684922
std       19.583905     3.517462
min        4.000000     1.000000
25%       35.000000     6.000000
50%       47.000000     8.000000
75%       62.000000    11.000000
max      100.000000    24.000000

#### With the above data, we see that the max words our titles have is 24 and the least is 1.
#### Using this, we can pad our titles with something at the end so that they are all the same length
##### Let's use "_"

In [25]:
TITLE_LEN = 24
def padTitle(title):
    titleLength = getSentenceLen(title)
    if  titleLength == 24:
        return title
    else:
        title = title + (" _" * (TITLE_LEN - titleLength))
        return title

In [26]:
titles.title = titles.title.apply(padTitle)
titles["word count"] = titles.title.apply(getSentenceLen)
titles["title length"] = titles.title.apply(len)

In [27]:
titles.describe()

title length  word count
count   5770.000000      5770.0
mean      80.161872        24.0
std       13.290563         0.0
min       50.000000        24.0
25%       71.000000        24.0
50%       79.000000        24.0
75%       88.000000        24.0
max      124.000000        24.0

In [28]:
data_cleaned.title = titles

##### There we are. Now we can vectorize the words using something like word2vec or gloVe

# -----------------------------------------------------------------------------------------------------------

####  Channel Title

For channel title, we can do something similar to Title

In [29]:
chans = data_cleaned.channel_title

In [30]:
chans.describe()

count     5770
unique    2056
top       ESPN
freq        70
Name: channel_title, dtype: object

#### Unlike video title, we want to keep the repeats in channel title. This is best explained in an example:

If a video by the The Tonight Show is what we're currently looking at, it is probably fair to assume
that Jimmy Fallon will be in the thumbnail.

***Interestingly, ESPN Takes the top spot for most trending videos in our data set***

In [31]:
chans

0                                CaseyNeistat
1                             LastWeekTonight
2                                Rudy Mancuso
3                       Good Mythical Morning
4                                    nigahiga
5                                    iJustine
6                         Saturday Night Live
7                          CrazyRussianHacker
8                            20th Century Fox
9                                         Vox
10                                        NFL
11                                        amc
12                                 marshmello
13                              NowThis World
14                            The king of DIY
15                             BostonDynamics
16                                    Cracked
17                                     TED-Ed
18                                PeterSripol
19                                    SciShow
20                                Life Noggin
21                                

# -----------------------------------------------------------------------------------------------------------

#### Category ID

Luckily enough, we can keep Category ID alone since it is already a nice number

In [32]:
cats = data_cleaned["category_id"]
cats

0        22
1        24
2        23
3        24
4        24
5        28
6        24
7        28
8         1
9        25
10       17
11       24
12       10
13       25
14       15
15       28
16       23
17       27
18       28
19       27
20       27
21       28
22       27
23       26
24       23
25       24
26       28
27       26
28       25
29        1
         ..
34556    27
34558    28
34750    26
34751    24
34753    10
34754    24
34755    25
34757    24
34758    24
34759    28
34760     1
34761    26
34765    19
34766    24
34768    25
34769    24
34770    28
34771    24
34772    24
34774    24
34776    10
34777    24
34781    24
34783    26
34798    17
34800    24
34801    10
34813    10
34816    25
34875    24
Name: category_id, Length: 5770, dtype: int64

# -----------------------------------------------------------------------------------------------------------

#### Tags

We can do almost exactly the same thing with tags that we did with title

In [33]:
tags = pd.DataFrame(data_cleaned.tags)

In [34]:
tags.describe()

tags
count     5770
unique    5317
top     [none]
freq       239

In [35]:
def getTagCount(s):
    l = s.split("|")
    c = 0
    c += sum([1 for t in l])
    return c

In [36]:
#redefined from above
def getSentenceLen(sent):
    tags = sent.split("|")
    counts = []
    for tag in tags:
        c = 0
        words = tag.split(" ")
        c += sum([1 for word in words])
        counts.append(c)
    return counts

In [37]:
tags["tag count"] = tags.tags.apply(getTagCount)
tags["tag word count"] = tags.tags.apply(getSentenceLen)

In [38]:
print(max(tags["tag word count"]))
print(min(tags["tag word count"]))

[16]
[1]


In [39]:
tags.describe()

tag count
count  5770.000000
mean     19.789081
std      12.241689
min       1.000000
25%      10.000000
50%      19.000000
75%      29.000000
max      69.000000

In [54]:
TAG_W_COUNT = 16
TAG_COUNT = 69
def padTags(s):
    tags = s.split("|")
    newTags = []
    for tag in tags:
        
        #Pad words
        words = tag.split(" ")
        if len(words) == TAG_W_COUNT:
            newTags.append(" ".join(words))
        else:
            while len(words) < TAG_W_COUNT:
                words.append("-")
            newTags.append(" ".join(words))
        
    while len(newTags) < TAG_COUNT:
        tmp = ""
        for i in range(TAG_W_COUNT-1):
            tmp += "- "
        
        newTags.append(tmp)
            
    return "|".join(newTags)

In [60]:
tags["tags"] = tags.tags.apply(padTags)
tags["tag count"] = tags.tags.apply(getTagCount)
tags["tag word count"] = tags.tags.apply(getSentenceLen)

In [62]:
data_cleaned.tags = tags.tags

In [63]:
data_cleaned

video_id trending_date  \
0      2kyS6SvSYSE      17.14.11   
1      1ZAPwfrtAFY      17.14.11   
2      5qpjK5DgCt4      17.14.11   
3      puqaWrEC7tY      17.14.11   
4      d380meD0W0M      17.14.11   
5      gHZ1Qz0KiKM      17.14.11   
6      39idVpFF7NQ      17.14.11   
7      nc99ccSXST0      17.14.11   
8      jr9QtXwC9vc      17.14.11   
9      TUmyygCMMGA      17.14.11   
10     9wRQljFNDW8      17.14.11   
11     VifQlJit6A0      17.14.11   
12     5E4ZBSInqUU      17.14.11   
13     GgVmn66oK_A      17.14.11   
14     TaTleo4cOs8      17.14.11   
15     kgaO45SyaO4      17.14.11   
16     ZAQs-ctOqXQ      17.14.11   
17     YVfyYrEmzgM      17.14.11   
18     eNSN6qet1kE      17.14.11   
19     B5HORANmzHw      17.14.11   
20     vU14JY3x81A      17.14.11   
21     6VhU_T463sU      17.14.11   
22     _-aDHxoblr4      17.14.11   
23     JBZTZZAcFTw      17.14.11   
24     lZ68j2J_GOM      17.14.11   
25     dRpNZV18N_g      17.14.11   
26     fcVjitaM3LY      17.14.11   
27     qeWvgZLz9yU      17.14.11   
28     iIxy3JN3-jc      17.14.11   
29     n30k5CwLhS4      17.14.11   
...            ...           ...   
34556  6ijnv-jNhUA      18.14.05   
34558  GAkkKjds00o      18.14.05   
34750  QBL8IRJ5yHU      18.15.05   
34751  t4pRQ0jn23Q      18.15.05   
34753  MAjY8mCTXWk      18.15.05   
34754  xhs8tf1v__w      18.15.05   
34755  E21NATEP9QI      18.15.05   
34757  1RZYOeQeIXE      18.15.05   
34758  WF82ABLw8s4      18.15.05   
34759  r-3iathMo7o      18.15.05   
34760  NBSAQenU2Bk      18.15.05   
34761  Xpv-sEKl1B4      18.15.05   
34765  96oKlWv5wSo      18.15.05   
34766  oRexsyztGS0      18.15.05   
34768  1U1u5aKU3AY      18.15.05   
34769  xTrwT0jSUg0      18.15.05   
34770  3g5O-kT9m8k      18.15.05   
34771  vY_5EvoL1Is      18.15.05   
34772  Dwc27Lsr1EY      18.15.05   
34774  D2mxKEa2xmA      18.15.05   
34776  BspHjvU11y4      18.15.05   
34777  nRc0kmOYgzQ      18.15.05   
34781  BfawmhUVXVo      18.15.05   
34783  mAfkkgw_-68      18.15.05   
34798  mkz1zoo15zI      18.15.05   
34800  LtBpvxdI1yQ      18.15.05   
34801  tjA7nAHOAww      18.15.05   
34813  4LlQwTgB5Rc      18.15.05   
34816  MdZmi9HGd9w      18.15.05   
34875  MEcqo5fctxI      18.15.05   

                                                   title  \
0      WE WANT TO TALK ABOUT OUR MARRIAGE_ _ _ _ _ _ ...   
1      The Trump Presidency: Last Week Tonight with J...   
2      Racist Superman | Rudy Mancuso, King Bach & Le...   
3      Nickelback Lyrics: Real or Fake?_ _ _ _ _ _ _ ...   
4      I Dare You: GOING BALD!?_ _ _ _ _ _ _ _ _ _ _ ...   
5      2 Weeks with iPhone X_ _ _ _ _ _ _ _ _ _ _ _ _...   
6      Roy Moore & Jeff Sessions Cold Open - SNL_ _ _...   
7      5 Ice Cream Gadgets put to the Test_ _ _ _ _ _...   
8      The Greatest Showman | Official Trailer 2 [HD]...   
9      Why the rise of the robots won’t mean the end ...   
10     Dion Lewis' 103-Yd Kick Return TD vs. Denver! ...   
11     (SPOILERS) 'Shiva Saves the Day' Talked About ...   
12     Marshmello - Blocks (Official Music Video)_ _ ...   
13     Which Countries Are About To Collapse?_ _ _ _ ...   
14     SHOPPING FOR NEW FISH!!!_ _ _ _ _ _ _ _ _ _ _ ...   
15     The New SpotMini_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ ...   
16     One Change That Would Make Pacific Rim a Class...   
17     How does your body know you're full? - Hilary ...   
18     HomeMade Electric Airplane_ _ _ _ _ _ _ _ _ _ ...   
19     Founding An Inbreeding-Free Space Colony_ _ _ ...   
20     How Can You Control Your Dreams?_ _ _ _ _ _ _ ...   
21     The Making of Hela's Headdress from Thor: Ragn...   
22     Is It Dangerous To Talk To A Camera While Driv...   
23     What $4,800 Will Get You In NYC | Sweet Digs H...   
24     Using Other People's Showers_ _ _ _ _ _ _ _ _ ...   
25     SPAGHETTI BURRITO VS SPAGHETTI BURRITO_ _ _ _ ...   
26     78557 and Proth Primes - Numberphile_ _ _ _ _ ...   
27     A Smart... MUG?! - Take apart a Heated Thermos...   
28     LeBron James admits he was rip

# -----------------------------------------------------------------------------------------------------------

In [43]:
import keras
from keras.layers import Conv2D, Activation, Dropout
from keras.models import Sequential

Using TensorFlow backend.
